tools/parity_scale_decoder/src/main.rs

```diff
@@ -0,0 +1,279 @@
+//! Parity Scale decoder tool for Iroha data types. For usage run with `--help`
+
+#![allow(clippy::print_stdout, clippy::use_debug, clippy::unnecessary_wraps)]
+
+use std::{collections::BTreeMap, fmt::Debug, fs, io, io::Write, path::PathBuf};
+
+use clap::Parser;
+use eyre::{eyre, Result};
+use iroha_data_model::prelude::*;
+use parity_scale_codec::Decode;
+
+/// Parity Scale decoder tool for Iroha data types
+#[derive(Debug, Parser)]
+#[clap(version, about, author)]
+enum Args {
+    /// Show all available types
+    ListTypes,
+    /// Decode type from binary
+    Decode(DecodeArgs),
+}
+
+#[derive(Debug, clap::Args)]
+struct DecodeArgs {
+    /// Path to the binary with encoded Iroha structure
+    binary: PathBuf,
+    /// Type that is expected to be encoded in binary.
+    /// If not specified then a guess will be attempted
+    #[clap(short, long = "type")]
+    type_id: Option<String>,
+}
+
+/// Function pointer to [`DumpDecoded::dump_decoded()`]
+///
+/// Function pointer is used cause trait object can not be used
+/// due to [`Sized`] bound in [`Decode`] trait
+pub type DumpDecodedPtr = fn(&[u8], &mut dyn Write) -> Result<(), eyre::Error>;
+
+/// Map (Type Name -> `dump_decode()` ptr)
+pub type DumpDecodedMap = BTreeMap<&'static str, DumpDecodedPtr>;
+
+/// Types implementing this trait can be decoded from bytes
+/// with *Parity Scale Codec* and dumped to something implementing [`Write`]
+pub trait DumpDecoded: Debug + Decode {
+    /// Decode `Self` from `input` and dump to `w`
+    ///
+    /// # Errors
+    /// - If decoding from *Parity Scale Codec* fails
+    /// - If writing into `w` fails
+    fn dump_decoded(mut input: &[u8], w: &mut dyn Write) -> Result<(), eyre::Error> {
+        let obj = <Self as Decode>::decode(&mut input)?;
+        #[allow(clippy::use_debug)]
+        writeln!(w, "{:#?}", obj)?;
+        Ok(())
+    }
+}
+
+impl<T: Debug + Decode> DumpDecoded for T {}
+
+fn main() -> Result<()> {
+    let args = Args::parse();
+
+    let map = generate_map();
+    let stdout = io::stdout();
+    let mut writer = io::BufWriter::new(stdout.lock());
+
+    match args {
+        Args::Decode(decode_args) => decode(decode_args, &map, &mut writer),
+        Args::ListTypes => list_types(&map, &mut writer),
+    }
+}
+
+macro_rules! generate_map {
+    ($($type:tt),* $(,)?) => {
+        #[allow(trivial_casts)]
+        DumpDecodedMap::from([
+            $((stringify!($type), <$type as DumpDecoded>::dump_decoded as DumpDecodedPtr)),*
+        ])
+    };
+}
+
+#[allow(clippy::too_many_lines)]
+fn generate_map() -> DumpDecodedMap {
```

Comment:
The inconvenience of having to add to this macro to me seems greatly exaggerated.